File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.util.Scanner;

public class Main {
	static int N,M;
	static int[] switches;
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		
		N = sc.nextInt();
		switches = new int[N+1];
		for(int i=1;i<=N;i++) {
			switches[i] = sc.nextInt();
		}
		M = sc.nextInt();
		for(int i=0;i<M;i++) {
			int gender = sc.nextInt();
			int index = sc.nextInt();
			solve(gender,index);
			
		}
		for(int i=1;i<=N;i++) {
			System.out.print(switches[i]+" ");
			if(i%20==0) {
				System.out.println();
			}
		}
	}
	static void solve(int g,int idx) {
		if(g == 1) {
			int tmp=idx;
			while(tmp<=N) {
				switches[tmp] = Math.abs(switches[tmp]-1);
				tmp+=idx;
			}
		}
		else {
			int i=1;
			switches[idx] = Math.abs(switches[idx]-1);
			while(idx+i<=N && idx-i>0 && switches[idx+i] == switches[idx-i]) {
				switches[idx-i] = Math.abs(switches[idx-i]-1);
				switches[idx+i] = Math.abs(switches[idx+i]-1);
				i++;
			}
		}
	}
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include <iostream>

using namespace std;
int N,M;
int arr[101];
void solve(int g, int idx){
    if(g==1){
        int tmp = idx;
        while(tmp<=N){
            arr[tmp] = 1-arr[tmp];
            tmp+=idx;
            
        }
    }
    else{
        int i = 1;
        arr[idx] = 1- arr[idx];
        while(idx-i>0 && idx+i<=N && arr[idx-i] == arr[idx+i]){
             arr[idx-i] = 1-arr[idx-i];
            arr[idx+i] = 1- arr[idx+i];
            i++;
        }
        
    }
}

int main(){
     ios_base::sync_with_stdio(0);
    cin.tie(0);cout.tie(0);
    cin>>N;
    for(int i=1;i<=N;i++){
        cin>>arr[i];
    }
    cin>>M;
    for(int i=0;i<M;i++){
        int gender,index;
        cin>>gender>>index;
        solve(gender,index);
    }
    for(int i=1;i<=N;i++){
        cout<<arr[i]<<" ";
        if(i%20==0)
            cout<<"\n";
    }
    
    
    return 0;
}